bam/bam_test.go

```diff
@@ -193,6 +194,27 @@ func BenchmarkRead(b *testing.B) {
 	f.Close()
 }
 
+func BenchmarkReadIndex(b *testing.B) {
+	if *findex == "" {
+		b.Skip("no index file specified")
+	}
+	for i := 0; i < b.N; i++ {
+		b.StopTimer()
+		f, err := os.Open(*findex)
+		if err != nil {
+			b.Fatalf("Open failed: %v", err)
+		}
+		b.StartTimer()
+
+		_, err = ReadIndex(f)
```

Comment:
Read the file once into the bytes.Buffer and then pass the bytes to ReadIndex through a bytes.Reader. This way, you don't benchmark file reading.